lndr-backend/src/Lndr/Handler/Credit.hs

```diff
@@ -195,6 +196,7 @@ calculateSettlementCreditRecord config cr@(CreditRecord _ _ amount _ _ _ _ _ uca
     in cr { settlementAmount = Just $ roundToMegaWei settlementAmountRaw
           , settlementBlocknumber = Just blockNumber
           }
+calculateSettlementCreditRecord _ cr@(CreditRecord _ _ _ _ _ _ _ _ _ (Just _) _ _) = cr
```

Comment:
This shouldn't be `(Just _)` in the pattern match, but `Nothing` like before. Or in any case you need a separate pattern-match with `Nothing` otherwise you will have non-exhaustive pattern-match issues when a nothing field is passed here.

If you want to match everything here, just use an underscore. I see now that it'll always be a `Just` since it's the `calculateSettlementCreditRecord` functions. To be clear, replace `(Just _)` with `_`.